src/compiler/checker.ts

```diff
@@ -24472,6 +24472,9 @@ namespace ts {
                                  return false;
                              });
             }
+            if (compilerOptions.target! >= ScriptTarget.ES2015) {

```

Comment:
It should also be disallowed when in `strict` mode. e.g.:

```ts
function bar() {
    "use strict";
    l: function f() { }  // Error
}
```

Reply:
We do all our handling of strict mode checks in the binder and not in the checker.